ang/crmCaseType.js

```diff
@@ -300,16 +308,25 @@
             });
           }
         }
-        // TODO The ids below really should use names not labels see
-        //  https://lab.civicrm.org/dev/core/issues/774
+        // TODO The `id` value is caught in between trying to support xml
+        // files where using the relType.id doesn't make it portable between
+        // installs and the xml doesn't have a concept of direction, but
+        // using name_a_b would conflict with it being used currently to set
+        // displaylabel elsewhere. So for now `id` is label.
         else {
           result.push({
+            xmlNameOtherDirection: relType.name_a_b,
+            xmlName: relType.name_b_a,
             text: relType.label_b_a,
             id: relType.label_a_b
           });
 
           if (!isBidirectionalRelationship) {
             result.push({
+              // This looks wrong but at least for matching up to
+              // everything else xmlName needs to be backwards too.
+              xmlNameOtherDirection: relType.name_b_a,
+              xmlName: relType.name_a_b,
```

Comment:
Similarly, this should be

```js
              xmlName: relType.name_b_a,
              xmlNameOtherDirection: relType.name_a_b,
```

Basically, the `xmlName` should always be the opposite direction of `text`, since the former is what is written to XML and the latter is what the user sees.